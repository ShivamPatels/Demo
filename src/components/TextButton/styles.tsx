import { StyleSheet } from "react-native";
import colors from "../../utils/colors";
import { screenWidth } from "../../utils/dimensions";

const styles = StyleSheet.create({
    button: {
        alignItems: "center",
        justifyContent: "center",
        borderRadius: 40,
        backgroundColor: colors.BG_BUTTON,
        width: screenWidth - 40,
    },
    title: {
        color: colors.TXT_BUTTON,
        paddingVertical: 12,
        fontSize: 17,
        fontWeight: "700",
    },
});

export default styles;
