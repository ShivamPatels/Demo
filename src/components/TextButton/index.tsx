import React from "react";
import {
    StyleProp,
    TextStyle,
    TouchableOpacity,
    ViewStyle,
    Text,
} from "react-native";
import colors from "../../utils/colors";
import styles from "./styles";

interface Props {
    // TouchableOpacity
    style?: StyleProp<ViewStyle>;
    activeOpacity?: number;
    disabled?: boolean;
    onPress?: () => void;
    // Text
    titleStyle?: StyleProp<TextStyle>;
    title: string;
}

const TextButton: React.FC<Props> = ({
    // TouchableOpacity
    style,
    activeOpacity,
    disabled,
    onPress,
    // Text
    titleStyle,
    title,
}) => {
    return (
        <TouchableOpacity
            style={[
                styles.button,
                style,
                disabled && { backgroundColor: colors.TEXT_GREY },
            ]}
            activeOpacity={activeOpacity}
            disabled={disabled}
            onPress={onPress}
        >
            <Text style={[styles.title, titleStyle]}> {title}</Text>
        </TouchableOpacity>
    );
};

export default TextButton;
