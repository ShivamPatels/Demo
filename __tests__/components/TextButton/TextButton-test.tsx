/**
 * @format
 */

import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import TextButton from "../../../src/components/TextButton";

it("renders correctly", () => {
    const tree = renderer.create(<TextButton title="Apple" />).toJSON();
    expect(tree).toMatchSnapshot();
});
