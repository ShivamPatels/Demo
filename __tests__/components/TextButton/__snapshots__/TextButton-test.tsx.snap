// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, 0.05)",
      "borderRadius": 40,
      "justifyContent": "center",
      "opacity": 1,
      "width": 710,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#898A8D",
          "fontSize": 17,
          "fontWeight": "700",
          "paddingVertical": 12,
        },
        undefined,
      ]
    }
  >
     
    Apple
  </Text>
</View>
`;
